file.go

```diff
@@ -912,30 +912,40 @@ func (s *filePageValueReaderState) init(columnType Type, column *Column, codec f
 		return fmt.Errorf("cannot read values from page of type %s", h.PageType())
 	}
 
-	repetitionLevelEncoding := pageHeader.RepetitionLevelEncoding()
-	definitionLevelEncoding := pageHeader.DefinitionLevelEncoding()
 	pageEncoding := pageHeader.Encoding()
-
-	s.repetitions.decoder = makeDecoder(s.repetitions.decoder, s.repetitions.encoding, repetitionLevelEncoding, repetitionLevels)
-	s.definitions.decoder = makeDecoder(s.definitions.decoder, s.definitions.encoding, definitionLevelEncoding, definitionLevels)
 	s.page.decoder = makeDecoder(s.page.decoder, s.page.encoding, pageEncoding, pageData)
-
-	s.repetitions.encoding = repetitionLevelEncoding
-	s.definitions.encoding = definitionLevelEncoding
 	s.page.encoding = pageEncoding
 
+	maxRepetitionLevel := column.maxRepetitionLevel
+	maxDefinitionLevel := column.maxDefinitionLevel
+	hasLevels := maxRepetitionLevel > 0 || maxDefinitionLevel > 0
+	if hasLevels {
```

Comment:
I noticed there was an easy optimization here, we can avoid creating the repetition and definition level decoders when dealing with required columns.